### ISSUE TITLE
incorrectly expects namedtuple second argument to be a tuple literal

### ISSUE THREAD

**Issue by foresto** [python/mypy#18824]
**Bug Report**

mypy incorrectly flags `namedtuple` instantiations with a non-literal tuple as the second argument.

**To Reproduce**

```python
#!/usr/bin/env python3
"mypy incorrect complaint reproducer"
from collections import namedtuple
Tup1 = namedtuple('Tup1', 'a b')
Tup2 = namedtuple('Tup2', Tup1._fields + ('c',))
```

**Expected Behavior**

No error.

**Actual Behavior**

    mypy-namedtuple-repro.py:5: error: List or tuple literal expected as the second argument to "namedtuple()"  [misc]
    Found 1 error in 1 file (checked 1 source file)

This complaint is incorrect. The `namedtuple` documentation says this about its second argument:

> The field_names are a sequence of strings such as ['x', 'y']. Alternatively, field_names can be a single string with each fieldname separated by whitespace and/or commas, for example 'x y' or 'x, y'.

Note that there is no requirement for that argument to be a list or tuple literal. Mypy is complaining about a perfectly valid construct.

**Your Environment**

https://mypy-play.net/

- Mypy version used: 1.15.0
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.12

**Comment by TeamSpen210** [comment]
At runtime you can indeed pass any sequence to `namedtuple`. But Mypy needs special handling for the function, so that it can identify fields and replicate the attributes and method signatures required. Without knowing the fields, it's going to be stuck treating instances as `Any`. This is kinda similar to doing like `class Indecisive(random.choice(list, str)):`, which would work at runtime but a checker has no real hope of handling. Python is super dynamic, the only way to handle the full extent of its behaviour is going to be by actually running the code.

If this particular behaviour was common enough, maybe it'd make sense to add some special case logic to specifically detect `NTClass._field + ('a', 'b', 'c')`. But I'm not sure if that's worth it. As an alternative, you could use dataclasses, which do support inheritance to build up their members.

**Comment by hauntsaninja** [comment]
It's actually fairly severe since mypy won't be able to check uses of this class correctly. If you're okay with that, this is what `# type: ignore` is for
